package.json

```
@@ -8,11 +8,11 @@
   },
   "dependencies": {
     "icss-replace-symbols": "1.1.0",
-    "postcss": "6.0.1",
-    "postcss-modules-extract-imports": "1.1.0",
-    "postcss-modules-local-by-default": "1.2.0",
-    "postcss-modules-scope": "1.1.0",
-    "postcss-modules-values": "1.3.0"
+    "postcss": "7.0.5",
```

I know this is not EECOLOR's fault – the PR is 1.5 years old – but pinning to 7.0.5 would cause the same issue I've pointed out.

This should have always been a semver range.

```suggestion
    "postcss": "^7.0.0",
```

---

As a side note, some versions before `7.0.5` contained bugs that prevented stuff from working. It's too long ago to remember which ones. If you still want to go this route I would suggest `^7.0.5`

---

It does not matter. By specifying a semver range, any dependents will  resolve the latest 7.x match.

In this case, everyone will be downloading 7.0.32 as that's currently the latest.
Pinning a specific version is a large part of what got `poatcss-modules` into this mess.

---

While you are right in most cases, if a user has `7.0.3` for example and he adds this module, things will break.

That said, practically I think it's fine to use `^7.0.0` because `7.0.5` was released long enough ago. The chance a user has a lower than `7.0.5` is very slim.

For these types of libraries I think I would even move the dependency to the `peerDependencies`.

---

Agreed, although I'm not sure if yarn@2 and pnpm provision peerDep access correctly. I assume they do, but not 100% sure.

In your example, if the user already had `7.0.3` locally, they'd _at least_ keep that. They certainly wouldn't do back down to `7.0.0`. But in 99% of cases, the fresh install of this module (assuming it ever gets out) triggers a refresh of its dependencies, resolving to the latest version that matches the specified range.